### ISSUE TITLE
Simplify steps for Create an Organization 

### ISSUE THREAD

**Issue by stefan-contiu** [Scille/parsec-cloud#1014]
We aim to simplify the UX for the first administrator that creates the organization. The goal is to start using Parsec as quick as possible.

Here are the new workflow steps :

1. Download Parsec from site vitrine.
2. In the Parsec Login UI, click "Create Organization".
3. A new window opens asking : Email Address, Organization Name, Device Name, Password, Repeat Password.
4. The administrator can start using Parsec right away (add data or add users)

In parallel with number 4, there is a second action flow :
5. The administrator receives an email with instructions on how to log-in to the site marchand.
6. Once logged in on the site marchand the administrator can (during a 14 days timeframe) :
6.1. Select a pricing plan (team or business)
6.2. Fill details of the account (company name, etc.)
6.3. Pay by using the credit card

**Comment by touilleMan** [comment]
Following today's meeting, here are the specs:

## 1 - SAAS Website quickjoin API

Should expose a new REST API:

`POST /api/quickjoin`

params fields:
- email
- organization_id

response fields:
- account_id: ID used by the SAAS api to refer to the created account
- bootstrap_link: organization bootstrap url

possible errors:
- `email_already_exists`
- `organization_already_exists`
- `internal_error` if backend doesn't respond

Under the hood, the api should:
1) create a new account into the SAAS database (db model should typically use inheritance to distinct between freshly created account and account with registration completed)
2) call the Parsec backend api to create a new organization 
3) send an email to notify the user of the account creation and provite him with a link to finalize the account (choose password, select plan etc.)

In case only part of the 3 steps are done (for instance if the server is shutdown) or if the client doesn't receive the response (bad internet connection) we endup with a bad situation: the client cannot reuse his email address given it is already taken by a half-finished process.
To solve this we should allow to re-do the process as long as the organization hasn't been bootstrapped.

## 2 - Parsec GUI quickjoin

On startup, if no devices are present, display a welcome screen with simple questions to determine whether the user wants to:
1) create a new device/user from an invitation
2) bootstrap an organization
3) create a new account in the SAAS website and a new organization
4) create a new organization from an existing SAAS account

Part 1&2 don't change from what currently exists.

Part 4 could be doable from the Parsec client (we would have to ask user email&password for the SAAS website, then use the REST api), however this is tricky in case the account hasn't been initialized yet (hence it doesn't have a password).
On top of that the SAAS password is easily mixed with the password encrypting device keys locally.
So the best thing to do is simply to redirect the user to the SAAS website login page and inform him about a confirmation email he must have received.

Part 3 should be implemented as a registration form with the following information:
- email,
- device_name (autofilled from computer name)
- organization_id: 
- password (must be typed twice to check for typo)

Once filled, the client should send a request to the SAAS website (see above) and receive the account_id and bootstrap url.
Bootstrap url is used to do the bootstrap step of the organization, account_id should be saved into the user manifest (a new `saas_account_id` optional field should be added to the user manifest).

## 3 - Access SAAS account from the Parsec GUI

Account_id is used to display an "access saas account" button in the GUI (typically in the settings menu). This button should open the browser on the SAAS website:
- If the user hasn't finalized it registering, he is invited to do so
- Otherwise the user endup on the login page with user login already filled

**Comment by Max-7** [comment]
In the errors that can be returned by the API, I would add
- invalid_email : the email is invalid
- invalid_organization_id : the organization name is invalid.
I can check those in the GUI before making the request but it would still be better to handle those cases correctly in the API.

**Comment by touilleMan** [comment]
One additional note: organization bootstrap requires a user_id
When #1011 will be merged we could use any random id, however for the moment it has to be readable enough.

The solution would be to sanitize the provided email, for instance take the part before the `@`, remove every invalid characters and limit to 32 characters:

```python
def email_to_user_id(email):
  part, _ = email.split('@', 1)
  return re.sub(r'[^\w\-]', '-', part)[:32]
```

Good thing is given we are bootstrapping the organization, no other user_id are present so we only need to have a valid user_id ;-)

**Comment by touilleMan** [comment]
Implemented by #1077
